libs/utils/android/workloads/geekbench.py

```diff
@@ -0,0 +1,155 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+from time import sleep
+
+from subprocess import Popen, PIPE
+
+from android import Screen, Workload, System
+
+class Geekbench(Workload):
+    """
+    Android Geekbench workload
+    """
+
+    # Package required by this workload
+    package = 'com.primatelabs.geekbench'
+    activity = '.HomeActivity'
+
+    def __init__(self, test_env):
+        super(Geekbench, self).__init__(test_env)
+        self._log = logging.getLogger('Geekbench')
+        self._log.debug('Workload created')
+
+    def run(self, out_dir, test_name, collect=''):
+        """
+        Run single Geekbench workload.
+
+        :param out_dir: Path on host to experiment directory where
+                        to store results.
+        :type out_dir: str
+
+        :param test_name: Name of the test to run
+        :type test_name: str
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+            - 'systrace'
+            - 'ftrace'
+            - any combination of the above in a space-separated string.
+        :type collect: list(str)
+        """
+
+        # Initialize energy meter results
+        self.out_dir = out_dir
+        self.collect = collect
+
+        # Regexps for benchmark synchronization
+        start_logline = r'ActivityManager: Start.*'\
+                         'com.primatelabs.geekbench'
+        GEEKBENCH_BENCHMARK_START_RE = re.compile(start_logline)
```

Comment:
I would have preferred to keep these "globals" defined outside the class... like we did for jankbench.
Otherwise using such a long name with "GEEKBENCH_BENCHMARK" as prefix does not make sense.

Unfortunately I just verified that I've merged your other workloads without pointing that out... so, it's up to you: either move them outside (like we do for jankbench) or just use "START_RE" and "END_RE".

I personally would prefer the first option which avoid to clobber the function code with funky REs :-)